src/libraries/System.Diagnostics.TraceSource/src/System/Diagnostics/TraceSwitch.cs

```diff
@@ -94,6 +95,13 @@ protected override void OnSwitchSettingChanged()
         protected override void OnValueChanged()
         {
             SwitchSetting = (int)Enum.Parse<TraceLevel>(Value, true);
+
+            Debug.Assert(Value.Length > 0, "Value.Length > 0");
+            if (char.IsDigit(Value[0]))
+            {
+                // Convert digit value representation to TraceLevel representation
+                SetSwitchValueString(((TraceLevel)SwitchSetting).ToString());
```

Comment:
I'm not following the logic here... why do we only do this if Value begins with a number?

Reply:
Yes, but `SetSwitchValueString` should also be also called if the user sets `Value` to "Off", "Error", "Warning", "Info" or "Verbose". With the code as it is right now, `SetSwitchValueString` will only be called if the user sets `Value` to "0", "1", "2", "3" or "4".

Related question: Why would we allow this?:
```cs
mySwitch.Value = "1";
```

if the user can set the number via the `Level`?
```cs
mySwitch.Level = 1;
```

Shouldn't we only allow `Value` to be set to "Off", "Error", "Warning", "Info" and "Verbose", and not the integer representation of their enum value?

Reply:
@carlossanlop `TraceSwitch` is wrapping the `TraceLevel` enum. If we restrict the `Value` to enume field names, this will be a breaking change and wouldn't be logical too as the runtime allows parsing numeric values.

Reply:
Okay, no need to restrict. But still, this needs fixing:

> Yes, but SetSwitchValueString should also be also called if the user sets Value to "Off", "Error", "Warning", "Info" or "Verbose". With the code as it is right now, SetSwitchValueString will only be called if the user sets Value to "0", "1", "2", "3" or "4".

Reply:
> SetSwitchValueString should also be also called if the user sets Value to "Off", "Error", "Warning", "Info" or "Verbose". With the code as it is right now, SetSwitchValueString will only be called if the user sets Value to "0", "1", "2", "3" or "4".

Users call `Value` property to set these values. 

https://github.com/dotnet/runtime/blob/587b0ad50e805e3467533e84e2bb3383f6b72824/src/libraries/System.Diagnostics.TraceSource/src/System/Diagnostics/Switch.cs#L183

For example, if you set the value to `Error`, this value will be stored in `_switchValueString` which same value returned from the property getter. So, there is no need to call `SetSwitchValueString` at that time because the right value is already there. If the user instead set the value to `1`, at that time `_switchValueString` will store this value. We'll need to change this value to `Error` and this is why I am calling `SetSwitchValueString` to reset `_switchValueString`.

Reply:
Ah okay, I see it now. Thanks for clarifying it.

Reply:
What happens if the user sets the value to a number that's not mapped to a named value?

What `char.IsDigit(Value[0])` is false because the text was padded with whitespace?

Reply:
I have addressed these fantastic questions with the latest commit [`985943c` (#89278)](https://github.com/dotnet/runtime/pull/89278/commits/985943c06b1e08b3938fa9da71e2a548d5703417). 

> What happens if the user sets the value to a number that's not mapped to a named value?

Now I am calling ` OnSwitchSettingChanged()` to let it normalize the value. This will make it match the behavior of the `Level` property. It is a good catch you raised it. 

> What if char.IsDigit(Value[0]) is false because the number was padded with whitespace?

I didn't know that `Enum.Parse` trim the white spaces when parsing. With the new change, I am calling `SetSwitchValueString` anyway to ensure always updating the string value. Thanks for the catch too.